internal/monitoring/responsewriter.go

```diff
@@ -0,0 +1,37 @@
+package monitoring
+
+import "net/http"
+
+// ResponseWriter wraps the http.ResponseWriter to give us the ability to
+// retrieve the HTTP status code at any point after the response has been written.
+type ResponseWriter struct {
+	http.ResponseWriter
```

Comment:
This hides all the other methods that the http ResponseWriter implementation implements that aren't part of the standard interface (e.g. Flusher, Hijacker, Pusher, CloseNotifier, ReaderFrom, StringWriter). If we don't care about any of them, then please add a comment to that effect, otherwise it might be worth passing through some of those interfaces explicitly.